tests/xfstests/run.pm

```diff
@@ -170,6 +176,45 @@ sub tests_from_category {
     return @tests;
 }
 
+# Return matched exclude tests from groups in @GROUPLIST
+# return structure - hash
+# Group name start with ! will exclude in test, and expected to use to update blacklist
+sub exclude_grouplist {
+    my %tests_list = ();
+    foreach my $group_name (@GROUPLIST) {
+        next if ($group_name !~ /^\!/);
+        $group_name = substr($group_name, 1);
+        my $cmd = "awk '/$group_name/ {printf \"$FSTYPE/\"}{printf \$1}{printf \",\"}' $INST_DIR/tests/$FSTYPE/group > tmp.group";
+        script_run($cmd);
+        $cmd = "awk '/$group_name/ {printf \"generic/\"}{printf \$1}{printf \",\"}' $INST_DIR/tests/generic/group >> tmp.group";
+        script_run($cmd);
+        $cmd = "cat tmp.group";
+        my %tmp_list = map { $_ => 1 } split(/,/, substr(script_output($cmd), 0, -1));
+        %tests_list = (%tests_list, %tmp_list);
+    }
+    return %tests_list;
+}
+
+# Return matched include tests from groups in @GROUPLIST
+# return structure - array
+# Group name start without ! will include in test, and expected to use to update test ranges
+sub include_grouplist {
+    my @tests_list;
+    foreach my $group_name (@GROUPLIST) {
+        next if ($group_name =~ /^\!/);
+        my $cmd = "cat $INST_DIR/tests/$FSTYPE/group | grep $group_name | awk '{printf \"$FSTYPE/\"}{printf \$1}{printf \",\"}' > tmp.group";
```

Comment:
If you could update cat | grep | awk  as it few lines above, it would be great. Thank you.

Reply:
fixed